app/collection.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/utils"
+)
+
+func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType string) error {
+	if !a.Config().FeatureFlags.ThreadsEverywhere {
+		return model.NewAppError("registerCollectionAndTopic", "app.collection.feature_flag.app_error", nil, "", http.StatusBadRequest)
+	}
+	// we have a race condition due to multiple plugins calling this method
+	a.ch.collectionAndTopicTypesMut.Lock()
+	defer a.ch.collectionAndTopicTypesMut.Unlock()
+
+	// check if collectionType was already registered by other plugin
+	for plugID, colTypes := range a.ch.collectionTypes {
```

Comment:
nit: perhaps `existingPluginID` and `existingCollectionTypes`, to better clarify the distinction?